pkgs/top-level/all-packages.nix

```diff
@@ -157,11 +40,7 @@ let
   #
   # The result is `pkgs' where all the derivations depending on `foo'
   # will use the new version.
-  overridePackages = f:
-    let
-      newpkgs = pkgsFun newpkgs overrides;
-      overrides = mkOverrides pkgs (f newpkgs pkgs);
-    in newpkgs;
+  overridePackages = f: pkgsWithOverrides f;
```

Comment:
Fair enough, this can be dealt with separately. The fact that nixpkgs even has such `~/.nixpkgs.config.nix` etc hardcodings kinda scares me, but at least it's cordoned from all-packages now.